content/ui-patterns/forms.mdx

```diff
@@ -0,0 +1,295 @@
+---
+title: Forms
+---
+
+import {Box, Text} from '@primer/components'
+
+<Box sx={{fontSize: 3}} class="lead" as="p">
+    Forms are used to complete tasks that require data input from the user. For example: creating a new repo, configuring settings, and logging in.
+</Box>
+
+<Box sx={{fontSize: 3}} class="lead" as="p">
+    Primer's form design guidelines aim to minimize the effort and cognitive load required to complete a task that involves a form.
+</Box>
+
+## Form control anatomy
+
+![diagrams labeling the anatomy of a text field and a checkbox field](https://user-images.githubusercontent.com/2313998/164065184-e3c615a3-fb68-4878-b7f8-9a57efb748cb.png )
+
+### Label (required)
+
+Labels should be descriptive, and as short as possible: ideally no more than 3 words. Labels should be written in sentence case. For example: "Repository name".
+
+If you're having trouble keeping label text short, consider using a [caption](#caption) to provide more context.
+
+In rare cases, a label may be hidden when the user has enough context to understand what the input does. When a label is visually hidden, label text must still be provided for assistive technology such as screen readers. For example: a search field with an icon that is placed in the spot a user would expect to find a search field may have a visually hidden label that says "Search".
```

Comment:
> In rare cases, a label may be hidden

> For example: a search field with an icon that is placed in the spot a user would expect to find a search field may have a visually hidden label that says "Search".

Is a search field part of a forms pattern? Search fields appear throughout dotcom, and they're not rare =).

Reply:
> Is a search field part of a forms pattern?

It _could_ be, but usually it isn't. I was trying to think of the most common and obvious example.

Do you (or any other reviewers) have any better examples?

Reply:
@mperrotti I'm not sure. It's confusing to see an example about a search field while this page is specifically about [long-form] forms pattern.

Reply:
Good point about these guidelines being for long-form forms. I think it might be best to exclude this paragraph. Visually hidden labels can be addressed in form **component** guidelines.